smart_open/tests/test_hdfs.py

```diff
@@ -31,23 +32,29 @@
 # such as Windows.
 #
 class CliRawInputBaseTest(unittest.TestCase):
+    def _setUp(self, test_file):
+        if platform == "linux" or platform == "linux2":
+            path = P.join(CURR_DIR, 'test_data/' + test_file)
+            self.cat = subprocess.Popen(['cat', path], stdout=subprocess.PIPE)
+        elif platform == "win32":
+            path = P.join(CURR_DIR, 'test_data\\' + test_file)
+            self.cat = subprocess.Popen(['type', path], stdout=subprocess.PIPE, shell=True)
+
     def test_read(self):
-        path = P.join(CURR_DIR, 'test_data/crime-and-punishment.txt')
-        cat = subprocess.Popen(['cat', path], stdout=subprocess.PIPE)
+        self._setUp(test_file='crime-and-punishment.txt')
```

Comment:
This is over-engineered. You don't need to use inheritance and OO for this stuff. Do this instead:

At module level:

```python
def cat(path):
    executable = 'cat'
    shell = False
    if sys.platform == 'win32':
        executable = 'type'
        shell = 'True'

    return subprocess.Popen([executable, path, stdout=subprocess.PIPE, shell=shell)
```

then in your tests:

```python
with mock.patch('subprocess.Popen', return_value=cat(path)):
```